hipposerve/api/product.py

```diff
@@ -274,6 +277,60 @@ async def confirm_product(
     logger.info("Successfully confirmed product {} (id: {})", item.name, item.id)
 
 
+@product_router.get("/{id}/set-visibility/{visibility}")
+async def set_visibility(
+    id: PydanticObjectId,
+    visibility: str,
+    calling_user: UserDependency,
+) -> dict:
+    """
+    Update the visibility of a product version.
+    Only owners or administrators can change visibility.
+    """
+
+    logger.info("Set visibility request for {} from {}", id, calling_user.name)
+
+    try:
+        visibility_enum = Visibility(visibility)
+        logger.info("Set visibility request for {} ", visibility)
```

Comment:
You should do this validation through pydantic and the body parameter deserialization